### PR TITLE
bitnet: honor relu2 hidden_activation from GGUF (fixes microsoft/BitNet#602)

### LINKED ISSUE #602: bitnet-b1.58-2B-4T runs FFN with SiLU instead of relu2 - wrong logits on every backend
# bitnet-b1.58-2B-4T runs the FFN with SiLU instead of relu² — wrong logits on every backend

## Summary

The official `microsoft/bitnet-b1.58-2B-4T` model declares `"hidden_act": "relu2"` in its `config.json`, but the BitNet.cpp runtime never reads it: the FFN graph is built with a hardcoded `LLM_FFN_SILU`. The model therefore runs with the wrong activation function on **every** backend (CPU and GPU), producing wrong-but-finite output — the same failure class as #586.

## Root cause

- `src/models/bitnet.cpp:133` — `build_ffn(...)` is called with a hardcoded `LLM_FFN_SILU`.
- `src/llama-model.cpp` — the `LLM_FFN_OP_TYPES_FROM_STRING` map has `"relu"` but no `"relu2"`, and `llm_ffn_op_type_from_string()` is never wired into the bitnet arch.
- `utils/convert-hf-to-gguf-bitnet.py` / `utils/convert-ms-to-gguf-bitnet.py` — neither converter writes `<arch>.hidden_activation`, so the existing plumbing (`llama_hparams::llm_ffn_op`, `LLM_KV_HIDDEN_ACT`, `LLM_FFN_RELU_SQR` in `src/llama-graph.cpp`) is never exercised for bitnet models.

The graph builder already implements `LLM_FFN_RELU_SQR` (relu → sqr, `src/llama-graph.cpp:1690`); with `LLM_FFN_PAR` gating it produces exactly `relu²(gate) * up`, matching the HF `BitnetMLP` (`act_fn(gate) * up`).

## Fix (attached patches)

1. `src/llama-model.cpp`: add `{ "relu2", LLM_FFN_RELU_SQR }` to `LLM_FFN_OP_TYPES_FROM_STRING`.
2. `src/models/bitnet.cpp`: read `<arch>.hidden_activation` in `load_arch_hparams` (mirroring `llama_model_modern_bert`), defaulting to `LLM_FFN_SILU` for backward compatibility with existing GGUF files; use `hparams.llm_ffn_op` in the graph.
3. Converters: write `add_hidden_act("relu2")`.

## Verification

- Convert `microsoft/bitnet-b1.58-2B-4T` with the patched converter; `gguf-dump` should show `bitnet_b158.hidden_activation = "relu2"`.
- Compare logits before/after: with the fix the model produces coherent text; without it, output is wrong-but-finite.
- Old GGUF files (no key) keep the previous SiLU behavior — no regression.

## Related

- #586 — same wrong-but-finite class (ARM i2_s layout).

### PR DESCRIPTION
**Pull request by msrovani**
Fixes microsoft/BitNet#602. The official bitnet-b1.58-2B-4T declares hidden_act: relu2 but the bitnet arch hardcodes LLM_FFN_SILU in build_ffn, producing wrong-but-finite logits on every backend.

Changes:
- llama-model.cpp: add "relu2" -> LLM_FFN_RELU_SQR to LLM_FFN_OP_TYPES_FROM_STRING
- models/bitnet.cpp: read <arch>.hidden_activation in load_arch_hparams (mirroring modern-bert), defaulting to LLM_FFN_SILU for backward compatibility with existing GGUF files; use hparams.llm_ffn_op in the graph

LLM_FFN_RELU_SQR with LLM_FFN_PAR gating produces exactly relu2(gate) * up, matching the HF BitnetMLP (act_fn(gate) * up).

### CODE REVIEW

**Comment by taronaeo** [comment]
Is this the same as https://github.com/ggml-org/llama.cpp/pull/25885#issuecomment-5026159503?

File: src/models/bitnet.cpp [hunk 1]
```diff
@@ -3,6 +3,14 @@
 void llama_model_bitnet::load_arch_hparams(llama_model_loader & ml) {
     ml.get_key(LLM_KV_ATTENTION_LAYERNORM_RMS_EPS, hparams.f_norm_rms_eps);
 
+    // The official bitnet-b1.58-2B-4T declares "hidden_act": "relu2".
+    // Default to SiLU for backward compatibility with existing GGUF files.
+    hparams.llm_ffn_op = LLM_FFN_SILU;
+    std::string hidden_act;
+    if (ml.get_key(LLM_KV_HIDDEN_ACT, hidden_act, false)) {
+        hparams.llm_ffn_op = llm_ffn_op_type_from_string(hidden_act, LLM_FFN_SILU);
```
Comment (CISC) [inline src/models/bitnet.cpp:11]: This is slightly problematic as it translates `silu` to `LLM_FFN_SWIGLU`.

See [1bitLLM/bitnet_b1_58-3B/config.json](https://huggingface.co/1bitLLM/bitnet_b1_58-3B/blob/main/config.json#L10).

When I flagged this earlier I don't think we came to any solution on how to distinguish this, so I think you will just have to check and change it back to `LLM_FFN_SILU`.

**Review by CISC — COMMENTED** [review]
Please update OP with our PR template.

File: src/models/bitnet.cpp (hunk 1 above)
Comment (CISC) [reply src/models/bitnet.cpp:11]: Maybe just adding a `gated` flag to `llm_ffn_op_type_from_string`?
